src/components/Calendar/Calendar.tsx

```diff
@@ -71,6 +71,17 @@ const CalendarStyling = styled.div`
     }
 `;
 
+const endDateInclusiveEvents = (endDateExclusiveEvents: CalendarEvent[]): CalendarEvent[] => {
+    return endDateExclusiveEvents.map((exclusiveEvent: CalendarEvent) => {
+        const cloneEndDate = new Date(exclusiveEvent.end);
+        if (exclusiveEvent.allDay) {
+            cloneEndDate.setDate(cloneEndDate.getDate() + 1);
+        }
+        const inclusiveEvent: CalendarEvent = { ...exclusiveEvent, end: cloneEndDate };
+        return inclusiveEvent;
+    });
+};
+
 const Calendar: React.FC<CalendarProps> = ({
     initialEvents,
```

Comment:
Refactored as suggested (use ref was used at first because if not it will break, but now it seems like it is not breaking because I edited to pass the date by value instead of reference and it works)

Reply:
how were you passing it 'by reference' before??

Reply:
I can't find the commit history of the version without cloning, but I think for this line: 

const cloneEndDate = new Date(exclusiveEvent.end);

I initially used: 

const cloneEndDate = exclusiveEvent.end;

So rather than creating a new Date object with the same date I passed in the original object that gets updated every time